polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -1618,7 +1619,13 @@ MLIRASTConsumer::getOrCreateLLVMFunction(const clang::FunctionDecl *FD) {
                                                 /*isVarArg=*/FD->isVariadic());
   // Insert the function into the body of the parent module.
   OpBuilder Builder(Module->getContext());
-  Builder.setInsertionPointToStart(Module->getBody());
+  if (FuncContext == FunctionContext::SYCLDevice)
+    Builder.setInsertionPointToStart(
+        mlirclang::getDeviceModule(*Module).getBody());
+  else {
+    assert(FuncContext == FunctionContext::Host);
+    Builder.setInsertionPointToStart(Module->getBody());
```

Comment:
NIT: Can we have a switch here with a `llvm_unreachable` in the default case instead?